usagov/vote-gov-drupal
# VOTE-1699: Update drupal core and dependencies.

## Jira ticket

[VOTE-1699](https://bixal-projects.atlassian.net/browse/VOTE-1699) - Update Drupal Core
[VOTE-1700](https://bixal-projects.atlassian.net/browse/VOTE-1700) - Update Drupal Modules

## Description

Update Drupal Core and its dependencies to 10.0.9 from 10.1.2.
Update Drupal modules:
- Metatag 1.23.0 - 2.0.0
- Simple Sitemap 4.1.4 - 4.1.6
- Token 1.11.0 - 1.12.0
- SAMLAuth 3.8.0 - 3.9.0
- S3FS 3.2.0 - 3.3.0
- Migration Tools 6.0.1 - 6.0.2
- Drush 11.5.1 - 11.6.0
- Node Revision Delete  1.0.0-rc6 - 2.0.0-alpha2
- Config Ignore 2.4.0 - 3.0.0-beta2
- Allowed Formats - 2.0.0 - 3.0.0-alpha1

## Deployment and testing

### Pre-deploy

1. Uninstall Config Ignore and Config Filter modules.
Could lead to missing files for Config Filter when config import runs.

### Post-deploy
For local deployment
1. run `lando composer install`
2. run `lando drush updatedb`

For cloud environments
1. Confirm that `drush updatedb` has been executed.

### QA/Test

1. Login as an Admin user 
3. Goto the Status page /admin/reports/status
4. Confirm that the Drupal Core version shows 10.1.2
5. Goto the Available updates page /admin/reports/updates
6. Confirm that there are no Security updates available and everything appears updated/current (green).
7. Goto Content Admin page
8. Goto the Edit page for each type of content (basic page, landing page, and State/Territory Page)
9. Confirm there are no visible errors.
10. Confirm that you can Save a content item without any visible errors.
11. Click around the front end of the site and confirm that site is still operational.
12. Goto the Recent Log Messages page /admin/reports/dblog
13. Confirm that there aren't any errors which stands out pertaining to any of the Updated modules above.

## Checklist for the Developer




## Checklist for the Peer Reviewers
